test/boost/schema_loader_test.cc

```diff
@@ -327,9 +354,39 @@ SEASTAR_TEST_CASE(test_load_schema_from_sstable) {
                 mr.consume_in_thread(std::move(wr));
             }
 
+            const auto sst_path = std::filesystem::path(fmt::to_string(sst->get_filename()));
+
             // Do the check in a separate method to ensure we don't accidentally
             // re-use the original schema.
-            check_sstable_schema(env, std::filesystem::path(fmt::to_string(sst->get_filename())), mutations);
+            check_sstable_schema(env, sst_path, mutations, true);
+
+            // Drop the scylla-metadata to also test the fall-back schema loader from Statistics
+            {
+                // rm Scylla.db
+                const auto scylla_metadata_path = fmt::to_string(sstables::component_name(*sst, component_type::Scylla));
+                remove_file(scylla_metadata_path).get();
+
+                // drop Scylla.db from TOC.txt
+
+                const auto toc_path = fmt::to_string(sstables::component_name(*sst, component_type::TOC));
+                auto toc_file = open_file_dma(toc_path, open_flags::rw).get();
+                auto ifstream = make_file_input_stream(toc_file);
+
+                auto toc = util::read_entire_stream_contiguous(ifstream).get();
+                const char scylla_component_name[] = "Scylla.db\n";
+                const auto scylla_component_name_pos = toc.find(scylla_component_name, 0, strlen(scylla_component_name));
+                BOOST_REQUIRE_NE(scylla_component_name_pos, sstring::npos);
+                toc.replace(scylla_component_name_pos, strlen(scylla_component_name), "", 0);
+
+                toc_file.truncate(0).get();
+
+                auto ofstream = make_file_output_stream(toc_file).get();
+                ofstream.write(toc).get();
+                ofstream.close().get();
+            }
+
+            check_sstable_schema(env, sst_path, mutations, false);
+
         }
```

Comment:
Would like to see a test that includes UDTs, collections, frozen/unfrozen.

Reply:
Random schema covers it all, but I will add a targeted test with more "interesting" types.

Reply:
Added additional test in v3.